programs/mango-v4/src/instructions/perp_liq_base_position.rs

```diff
@@ -96,41 +128,42 @@ pub fn perp_liq_base_position(
     let liqor_perp_position = liqor
         .ensure_perp_position(perp_market_index, perp_market.settle_token_index)?
         .0;
-    let liqee_base_lots = liqee_perp_position.base_position_lots();
 
     // Settle funding
     liqee_perp_position.settle_funding(&perp_market);
     liqor_perp_position.settle_funding(&perp_market);
 
+    // Max settleable on the liqee?
+    let now_ts: u64 = Clock::get()?.unix_timestamp.try_into().unwrap();
+    liqee_perp_position.update_settle_limit(&perp_market, now_ts);
+    let liqee_positive_settle_limit = liqee_perp_position.available_settle_limit(&perp_market).1;
+    let max_settle = I80F48::from(max_quote_transfer)
+        .min(liqor_settle_health)
```

Comment:
note to self: not obvious why this is used here

Reply:
I think it's not strictly necessary in this case.

For normal settlement it's essential because settlement is health-neutral on the account with negative pnl, but converts negative perp health to negative spot health -- and we don't want to allow converting perp losses into unbacked spot borrows this way.

In this case, the liqor pays `1 - liq_fee` spot for every 1 quote he takes over, so for each unit of perp pnl taken over, his health changes by `overall_perp_asset_weight - (1 - liq_fee) < 0`. That automatically limits the amount a liqor can take on.